datalad_next/utils/deprecate.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+import warnings
+from functools import wraps
+
+from datalad_next.utils import get_wrapped_class
+
+
+def deprecated(msg, version, parameter=None, parameter_choice=None):
```

Comment:
```suggestion
def deprecated(msg, *, version, parameter=None, parameter_choice=None):
```

I presently consider it better to make the nature of the version declaration more obvious in the resulting code. It is already mandatory (which is essential, I think).

A custom message is also useful to be mandatory. Deprecation without guidance on adjustment is bad practice.

Given that there are two essential items, it would be great to have the different semantics be obvious in each declaration.